Feature : Add only date as well as date-time section in Widget

Priyammm05 opened the pull request:
**What does this PR do?**

Fixes #6 

I have added only date section

https://clipchamp.com/watch/bIZRqwNDDGM

**Type of change:** 

- Added DigitalDate section in the Widgets
- Added different dateFormats and Separator for DigitalDate.

BirjuVachhani commented:
@Priyammm05 Thanks for the PR. Will try to review it today!

BirjuVachhani commented on lib/home/widgets/digital_date_widget.dart:
```diff
@@ -0,0 +1,86 @@
+        return 'MM${separator}dd${separator}yyyy';
+      case DateFormat.yearMonthDay:
+        return 'yyyy${separator}MM${separator}dd';
+      case DateFormat.special:
+        return 'dd $separator MMMM $separator yyyy';
```
This is very limiting. We can have a custom option where we can add an input field instead that allows to specify a DateFormat string that we can use directly! We can validate this by parsing current date and see if its a valid string.

https://api.flutter.dev/flutter/intl/DateFormat-class.html

Priyammm05 replied:
Ok so you means that when user click on special(custom) dropdown we should show a input box where user can give any string format like yyyy-MM-dd and base on that we should render date right ?

BirjuVachhani replied:
Yes! You can name it custom instead of Special.

BirjuVachhani commented on lib/home/widgets/digital_date_widget.dart:
```diff
@@ -0,0 +1,86 @@
+        return 'yyyy${separator}MM${separator}dd';
+      case DateFormat.special:
+        return 'dd $separator MMMM $separator yyyy';
+      case DateFormat.random:
+        final components = ['dd', 'MM', 'yyyy']..shuffle(random);
+        return components.join(separator);
```
This doesn't make much sense and doesn't seem that useful. Most people have a preference about this so this is most likely gonna be unnecessary. What do you think?

BirjuVachhani commented on lib/settings/widget_settings/digital_date_widget_settings.dart:
```diff
@@ -0,0 +1,109 @@
+          builder: (context) {
+            return CustomDropdown<DateFormat>(
+              isExpanded: true,
+              value: settings.format,
```
This shows `dayMonthYear` in the dropdown which not good because people are not used to this. They are used to `dd/mm/yyyy` format. We have separator info we can render these with that.

Priyammm05 replied:
Actually I have written this as `dayMonthYear` because if we write `dd/mm/yyyy` like this because if we use this `dd/mm/yyyy` then it already contains separator and I think that it limits when user have to change options.

BirjuVachhani replied:
We can use the separator selected by the user instead of /. It doesn't matter that much since its a template of how it would look like! But showing dayMonthYear might be more confusing.

BirjuVachhani commented:
@Priyammm05 We should probably add option to allow time with date since we are using DateFormat which can support time too!

Priyammm05 commented:
@BirjuVachhani adding date is good but it actually conflict the descriptions of application which is `minimal` as I have tried it and because of which I haven't added. If you think we should add I can add it.

BirjuVachhani commented:
> @BirjuVachhani adding date is good but it actually conflict the descriptions of application which is `minimal` as I have tried it and because of which I haven't added. If you think we should add I can add it.

I think adding time with date would not affect minimal aspect of the app. It should be fine! We can allow doing so with custom templates only.

Priyammm05 commented:
Something like this ?

![image](https://github.com/BirjuVachhani/pluto/assets/80326453/9c0c5522-d970-463d-906c-61de76c20068)

BirjuVachhani commented:
It can be on 1 line like this...
Also, no need to specify a separator for time! Time option should be available via custom option only I think.

<img width="154" alt="image" src="https://github.com/BirjuVachhani/pluto/assets/20423471/93006f8e-4660-4c5f-b77a-2ce85793363e">

Priyammm05 commented:
So no separator for date and time and it should be one liner ?

Priyammm05 commented:
Can you please just justify what are you expecting so that I can make all the changes at once.

BirjuVachhani commented:
Sure...

1. Only have a single separator dropdown which applies to date formats.
2. When date format type is custom, ignore the separator dropdown value and let user specify separator in the input field along with the template. This template also allows time with date and whatever the user has specified in this input field. the time should appear as it is. No extra separator selection for it!

Priyammm05 commented:
1. Added Date Format in row with single separator which applies to date format
![image](https://github.com/BirjuVachhani/pluto/assets/80326453/aa9a30da-0b86-456f-a9b5-294be473f6b0)

2. Added the custom type where if the user select the special, the user can input the template in which they want to see the date and time, or only date or only time with no extra separator
![image](https://github.com/BirjuVachhani/pluto/assets/80326453/5b67e5cc-91e8-4b8d-8dc0-1bfe0bebac57)

BirjuVachhani commented:
Looks good. 👍

Priyammm05 commented:
If everything good then can you merge it ?

BirjuVachhani commented:
LGTM
